lib/model.js

```diff
@@ -79,7 +79,9 @@ module.exports = class Model extends EventEmitter {
     if (typeof reqConf === 'string') {
       reqConf = url.parse(reqConf);
     }
-    return reqConf;
+    return Object.assign({}, reqConf, {
+      headers: options.headers || reqConf.headers || this.options.headers
```

Comment:
Is there any way for `reqConf.headers` to exist here? I was under the impression that was the point of the change.